addons/crm/static/src/views/crm_form/crm_form.js

```diff
@@ -61,7 +61,7 @@ export class CrmFormController extends FormController {
             localData._changes = changes;
         }
         let changedStage = false;
-        if ("stage_id" in changes) {
+        if ("stage_id" in changes && this.model.root.resId) {
```

Comment:
Hello 😄 
I got told of the issue and made a quick fix over here: https://github.com/odoo/odoo/pull/100863
My method is slightly different and allows newly created leads to also trigger the rainbowman message.
I'll let you check with the PO which is the preferred way :)
Cheers

For technicalities:
`bm.localData[recordID].data` is meant to be the state of the record as it is on the database, meaning it's empty for new records.